src/imgix-core-js.js

```diff
@@ -1,103 +1,126 @@
-import md5 from "js-md5";
-import URI from "urijs";
-import Base64 from "js-base64";
-
-export const VERSION = "0.3.1";
-
-export class Path {
-  constructor(path, host, token=null, secure=true, librarySignature="js", libraryVersion=VERSION) {
-    this.path = path;
-    this.host = host;
-    this.token = token;
-    this.secure = secure;
-    this.queryParams = {};
-    this.librarySignature = librarySignature;
-    this.libraryVersion = libraryVersion;
-
-    // We are dealing with a fully-qualified URL as a path, encode it
-    if (this.path.indexOf("http") === 0) {
-      this.path = URI.encode(this.path);
-    } else {
-      this.path = URI.encodeReserved(this.path);
-    }
-
-    if (this.path[0] !== "/") {
-      this.path = "/" + this.path;
-    }
+(function (global, factory) {
+  if (typeof define === 'function' && define.amd) {
+    define('Imgix', ['exports', 'js-md5', 'js-base64'], factory);
+  } else if (typeof exports !== 'undefined') {
+    module.exports = factory(exports, require('js-md5'), require('js-base64'));
+  } else {
+    var mod = {
+      exports: {}
+    };
+    global.ImgixClient = factory(mod.exports, global.md5, global.Base64);
   }
+})(this, function (exports, _jsMd5, _jsBase64) {
+  var md5 = _jsMd5;
+  var Base64 = _jsBase64;
+
+  var VERSION = '0.3.1';
+  var DEFAULTS = {
+    host: null,
+    useHTTPS: true,
+    includeLibraryParam: true
+  };
+
+  var ImgixClient = (function() {
+    function ImgixClient(opts) {
+      var key, val;
+
+      this.settings = {};
+
+      for (key in DEFAULTS) {
+        val = DEFAULTS[key];
+        this.settings[key] = val;
+      }
 
-  toString() {
-    let uri = new URI({
-      protocol: this.secure ? "https" : "http",
-      hostname: this.host,
-      path: this.path,
-      query: this._query()
-    });
-    return uri.toString();
-  }
+      for (key in opts) {
+        val = opts[key];
+        this.settings[key] = val;
+      }
 
-  toUrl(newParams) {
-    this.queryParams = Object.assign(this.queryParams, newParams);
-    return this;
-  }
+      if (!this.settings.host) {
+        throw new Error('ImgixClient must be passed a valid host');
+      }
 
-  _query() {
-    return URI.buildQuery(Object.assign(this._queryWithoutSignature(), this._signature()), false, false);
```

Comment:
I 100% agree with this in almost every case. However, with the imgix libraries, I'd like to try to avoid third-party URL building libraries, since the imgix libraries themselves are effectively glorified URL building libraries.